File: deployment/docker/backend/models/agent.py
from pydantic import BaseModel
from typing import Optional, List

class Agent(BaseModel):
    id: str
    name: str
    description: Optional[str] = None
    specialty: Optional[str] = None
    category: str = "general"
    is_premium: bool = False
    is_safe: bool = False
    icon: str = "🤖"
    prompt_template: Optional[str] = None
    tags: List[str] = []

File: apps/android/app/src/main/java/com/agi/platform/domain/model/Agent.kt
package com.agi.platform.domain.model

data class Agent(
    val id: String,
    val name: String,
    val description: String,
    val specialty: String,
    val category: String,
    val isPremium: Boolean = false,
    val isSafe: Boolean = false,
    val icon: String? = null
)

data class ChatMessage(
    val id: String,
    val agentId: String,
    val text: String,
    val isUser: Boolean,
    val timestamp: Long = System.currentTimeMillis()
)
